cluster-state-service/internal/features/e2einstancesteps/list_instances_steps.go

```diff
@@ -73,10 +73,32 @@ func init() {
 	})
 
 	Then(`^the list instances response contains all the instances registered with the cluster$`, func() {
-		if len(ecsContainerInstanceList) != len(cssContainerInstanceList) {
+		cssContainerInstanceARNs := make([]*string, 0, len(cssContainerInstanceList))
+		for _, v := range cssContainerInstanceList {
```

Comment:
Please also add a comment explaining we look at ACTIVE statuses because of CSS's retains inactive instances till reconciler re-runs and we want to avoid that dependency.